### PR TITLE
refactor: merge osd into machined

### PR DESCRIPTION
**Pull request by smira**
This merges `osd` API into `machined`. API was copied from `osd` into
`machined`, and `osd` API was deprecated.

For backwards compatibility, `machined` still implements `osd` API, so
older Talos API clients can still talk to the node without changes.

Docs were updated. No functional changes.

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/2303)

### CODE REVIEW

File: internal/app/machined/internal/server/v1alpha1/osd_compat.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+)
+
+type osdServer struct {
+	*Server
+}
+
+// Dmesg implements the osapi.OsServer interface.
+func (s *osdServer) Dmesg(req *machine.DmesgRequest, srv os.OSService_DmesgServer) error {
+	return s.Server.Dmesg(req, machine.MachineService_DmesgServer(srv))
+}
```
[the remark this answers is not in the document]
Comment (smira) [reply internal/app/machined/internal/server/v1alpha1/osd_compat.go:19]: This is the way Go & protoc work. The signature of `Dmesg` function is slightly different for `MachineService` and `APIService`:

```
 Dmesg(req *machine.DmesgRequest, srv os.OSService_DmesgServer)
```

and 

```
 Dmesg(req *machine.DmesgRequest, srv machine.MachineService_DmesgServer)
```

Even though interfaces are equivalent or identical, they are different types, so we can't have two `Dmesg` implementations in one `struct` nor we can have both interfaces satisfied with single struct.

Only a single method is "overridden" here to fix that (as it's the only streaming method).

[the remark this answers is not in the document]
Comment (andrewrynhard) [reply internal/app/machined/internal/server/v1alpha1/osd_compat.go:19]: :+1:

**Pull request merged**